roqoqo/build.rs

```diff
@@ -16,13 +16,24 @@ use std::fs;
 use std::path::PathBuf;
 use std::process::Command;
 use std::str::FromStr;
+use std::sync::{Mutex, OnceLock};
 use syn::parse::{Parse, ParseStream};
 use syn::punctuated::Punctuated;
 use syn::visit::{self, Visit};
 use syn::{AttrStyle, File, Ident, ItemImpl, ItemStruct, LitStr, Path, Token, Type, TypePath};
 
 const NUMBER_OF_MINOR_VERSIONS: usize = 16;
 
+static AVAILABLE_GATES: OnceLock<Mutex<Vec<String>>> = OnceLock::new();
+
+fn push_available_gate(gate: String) {
+    AVAILABLE_GATES
+        .get_or_init(|| Mutex::new(vec![]))
+        .lock()
+        .unwrap()
```

Comment:
Is this unwrap safe?

Reply:
If it weren't, it would mean that a second thread would be asking for the list while another one already has it locked. But I don't see a reason for that to happen in the `build.rs`.

Reply:
That makes sense, could you change this to `expects` with that exact message?